components/sup/src/http_gateway.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::net::{Ipv4Addr, SocketAddrV4};
+use std::ops::{Deref, DerefMut};
+use std::str::FromStr;
+use std::sync::{mpsc, Arc, RwLock};
+use std::thread::{self, JoinHandle};
+
+use butterfly;
+use hcore::package::{Identifiable, PackageIdent};
+use hcore::service::ServiceGroup;
+use iron;
+use iron::prelude::*;
+use iron::status;
+use iron::typemap;
+use persistent;
+use router::Router;
+use rustc_serialize::json;
+
+use config::gconfig;
+use error::{Result, Error, SupError};
+use health_check;
+use manager::{self, census};
+use util::signals;
+
+static LOGKEY: &'static str = "HG";
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct ListenAddr(pub SocketAddrV4);
+
+impl Default for ListenAddr {
+    fn default() -> ListenAddr {
+        ListenAddr(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), 9631))
```

Comment:
This will inline so we don't need to move it to a constant. This is literally the "default ip addr/port" for the http_gateway so I don't think this should be configurable here, instead we configure by the CLI & assign it to the Config struct. This is just what will be populated into that config struct by default